src/components/drawing-widget/drawing-widget.component.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect, useCallback } from "react";
+import ReactImageAnnotate, { Annotation } from "react-image-annotate";
+import { Add, Crop } from "@carbon/react/icons";
+import { useTranslation } from "react-i18next";
+import { CardHeader } from "@openmrs/esm-patient-common-lib";
+import { Button } from "@carbon/react";
+
+
+interface RegionData {
+  type: string;
+  x: number;
+  y: number;
+}
+
+export interface ImageData {
+  src: string;
+  name: string;
+  regions: RegionData[];
+}
+
+interface DrawingWidgetProps {
+  selectedImage: string;
+  taskDescription: string;
+  imagesData: ImageData[];
+  regionClsList?: string[];
+  enabledTools?: string[];
+  onExit: (annotations: ImageData[]) => void;
+}
+
+const DrawingWidget: React.FC<DrawingWidgetProps> = ({ onExit }) => {
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const [selectedImage] = useState<string | null>(null);
+  const [annotations, setAnnotations] = useState<ImageData[]>([]);
+  const [, setLoading] = useState<boolean>(true);
+  const [activeImage, setActiveImage] = useState<ImageData | null>(null);
+  const { t } = useTranslation();
+ 
+
+  const CreatePointIcon = () => <Add />;
+  const CreateBoxIcon = () => <Crop />;
+
+  useEffect(() => {
+    const selectedImageURLParam = new URLSearchParams(
+      window.location.search
+    ).get("image-url");
+    if (selectedImageURLParam) {
+      const activeImage: ImageData = {
+        src: selectedImageURLParam,
+        name: "Image from URL",
+        regions: [],
+      };
+      setActiveImage(activeImage);
+      setLoading(false);
+    } else {
+      setLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (activeImage) {
+      const initialAnnotations: ImageData[] = [
+        {
+          ...activeImage,
+          regions: activeImage.regions || [],
+        },
+      ];
+      setAnnotations(initialAnnotations);
+    }
+  }, [activeImage]);
+
+  const handleExit = useCallback(() => {
+    onExit(annotations);
+  }, [onExit, annotations]);
+
+  const handleAnnotationChange = useCallback(
+    (newAnnotations: Annotation[]) => {
+      setAnnotations((prevAnnotations) =>
+        prevAnnotations.map((prevAnnotation, index) =>
+          index === 0
+            ? { ...prevAnnotation, regions: newAnnotations }
+            : prevAnnotation
```

Comment:
I don't understand how the annotation data structure works, but this is strange. It appears that you only ever change the first element of `annotations`. There's nowhere for subsequent elements to come from. So why is it an array in the first place? And that first element, it just has `src`, `name`, and `regions`, and `regions` is the only thing that ever changes, and it's simply set by the annotation component. I think the data structure could be clarified a lot, and then your setters would be much simpler.